### DISCUSSION TITLE
What is polling switch?

### DISCUSSION THREAD

**Discussion by k2xl** [alandtse/tesla#90] [Q&A] [answered]
I see this switch but unsure what it does?

Also, what is the guidance of polling interval? Is this a basic polling interval or is there any special heuristics to reduce polling/vampire drain?

**Comment by alandtse** [comment, marked as answer]
When polling switch is enabled, the car will be polled based on the `scan interval`. The default is 660 seconds which is the same as the `sleep interval` but it depends on the car state.

- Car is driving - 60 seconds
- Car has switched to park - Greater of `sleep interval` or `scan interval` - This will allow a parked car to sleep.  This state will be skippedif the car is charging, sentry mode is on, or the climate system is on.
- All other times - Scan interval

If you turn off the switch, it will stop polling of the car entirely. You may want this if you're parked at an airport and you don't want the heuristics at all.

https://github.com/zabuldon/teslajsonpy/blob/dev/teslajsonpy/controller.py#L725-L778

**Comment by JevVan** [comment]
How can I set the values of the scan interval?  I'm trying to open the garage when I turn off the parking brake.   I'm not sure yet, if that's when I press the brake or put the car in gear.  Doesn't really matter, however, when I depress the brake would be best if that is an available sensor.

Back to the scan interval, I figure I can use the motion detector in the garage to reduce the scan interval down to .5-1 seconds for a couple minutes then put it back.  In the meantime, if the parking brake is turned off then I will open the garage.

I tried turning the poling switch off and on.  This seemed to work but it is not reliable.   Seems I need to set polling_policy to "always" but I don't see a switch to accomplish this.

**Reply by JevVan** [reply]
NM:  I found my own answer:  https://github.com/alandtse/tesla/wiki/Polling-policy
